toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1411,6 +1502,49 @@ func formatCycleErrorMessage(cycle []*PkgNode, err error) error {
 	return fmt.Errorf("cycles detected in dependency graph")
 }
 
+// NodesProvidedBySRPM returns all RPMs produced from a SRPM file.
+func NodesProvidedBySRPM(srpmPath string, pkgGraph *PkgGraph, graphMutex *sync.RWMutex) (rpmNodes []*PkgNode) {
+	if graphMutex != nil {
+		graphMutex.RLock()
+		defer graphMutex.RUnlock()
+	}
+
+	/*Get count of the number of rpms provided by the SRPM*/
```

Comment:
I'm curious if this is worth the speedup, I'd say each package probably has ~10 nodes (order of magnitude), while the graph has ~10000 nodes. I'm not sure pre-allocating the slice is worth the extra code/time complexity here.

Reply:
right, will pre-allocate few nodes, append will increase the size dynamically if required.